optuna/study/_multi_objective.py

```diff
@@ -127,17 +126,22 @@ def _is_pareto_front_nd(unique_lexsorted_loss_values: np.ndarray) -> np.ndarray:
     loss_values = unique_lexsorted_loss_values[:, 1:]
     n_trials = loss_values.shape[0]
     on_front = np.zeros(n_trials, dtype=bool)
-    nondominated_indices: np.ndarray[tuple[int, ...], np.dtype[np.signedinteger[Any]]] = np.arange(
-        n_trials
-    )
+    # TODO(nabenabe): Replace with the following once Python 3.8 is dropped.
+    # nondominated_indices: np.ndarray[tuple[int], np.dtype[np.signedinteger]] = ...
+    nondominated_indices = np.arange(n_trials)
```

Comment:
The original type `np.ndarray[tuple[int, ...], np.dtype[np.signedinteger[Any]]]` is not incorrect, just loosely typed, so I think that leaving a TODO comment would be better than using a type ignore statement.

Reply:
Done!